internals/overlord/overlord.go

```diff
@@ -379,15 +371,34 @@ func (o *Overlord) Loop() {
 			}
 		}
 	})
+	o.ensureWaitRun()
 }
 
 func (o *Overlord) ensureDidRun() {
-	atomic.StoreInt32(&o.ensureRun, 1)
+	select {
+	case <-o.ensureRun:
+		// Already closed. Ensure already ran at least once.
+	default:
+		close(o.ensureRun)
+	}
 }
 
-func (o *Overlord) CanStandby() bool {
-	run := atomic.LoadInt32(&o.ensureRun)
-	return run != 0
+// ensureWaitRun waits until StateEngine.Ensure() was called at least once.
+func (o *Overlord) ensureWaitRun() {
+	select {
+	case <-o.ensureRun:
+	case <-o.loopTomb.Dying():
+	}
+}
+
+func (o *Overlord) CanStandby() (ensured bool) {
+	select {
+	case <-o.ensureRun:
+		// Already closed. Ensure already ran at least once.
+		ensured = true
+	default:
+	}
+	return ensured
```

Comment:
The named return seems confusing here. What about:
```suggestion
func (o *Overlord) CanStandby() bool {
	select {
	case <-o.ensureRun:
		// Already closed. Ensure already ran at least once.
		return true
	default:
		return false
	}
```

Reply:
Improved as proposed. Resolving.